src/tests/EShopworld.WorkerProcess.UnitTests/LeaseAllocatorTests.cs

```diff
@@ -204,7 +204,7 @@ public async Task TestAllocateLeaseAsync_OnLeaseUpdateConflict_ShouldNotUpdateLe
         }
 
         [Fact, IsUnit]
-        public async Task TestAllocateLeaseAsync_OnLeaseCreateConflict_ShouldNotAcquireLease()
+        public async Task TestAllocateLeaseAsync_OnExistingLeaseWithSameInstanceId_ShouldNotAcquireLease()
```

Comment:
Should we create a brand new test for this scenario, while keeping the old one?

Reply:
The old one would fail with the current changes. So that's why I reused this one.

Reply:
True, but what i would expect is:

- Modifying existing test having the second read returning a lease with a different instanceId, so the old scenario will still apply

- Adding your test to verify the case when the instanceId is shared. 
Makes sense?